pages/profile/read/views/pil-apply.jsx

```diff
@@ -1,36 +1,41 @@
-import { connect } from 'react-redux';
 import React from 'react';
 import {
   Link,
   Snippet
 } from '@asl/components';
 
-const PilApply = ({
-  isUser,
-  allowedActions,
-  pil
-}) => {
-  if (!isUser && !allowedActions.includes('pil.create')) {
-    return null;
-  }
-  if (pil && pil.status === 'active') {
-    return null;
-  }
-  const incomplete = pil && pil.status !== 'active';
-  return (
-    <div>
-      <p><Snippet>{`pil.${isUser ? 'user' : 'other'}.${incomplete ? 'incomplete' : 'notStarted'}`}</Snippet></p>
-      <p>
-        <Link
-          page='pil.create'
-          className="govuk-button"
-          label={<Snippet>{`buttons.${incomplete ? 'continue' : 'applyNow'}`}</Snippet>}
-        />
-      </p>
-    </div>
-  );
-};
+class PilApply extends React.Component {
 
-const mapStateToProps = ({ static: { isUser, allowedActions } }) => ({ isUser, allowedActions });
+  render() {
+    const
+      {
+        allowedActions,
+        pil,
+        estId,
+        profileId
+      } = this.props;
+
+    if (!allowedActions.includes('pil.create')) {
+      return null;
+    }
+    if (pil && pil.status === 'active') {
+      return null;
+    }
+    const incomplete = pil && pil.status !== 'active';
+    return (
+      <div>
+        <p>
+          <Link
+            page='pil.create'
+            establishmentId={estId}
+            profileId={profileId}
+            className="govuk-button"
+            label={<Snippet>{`buttons.${incomplete ? 'continue' : 'pilApply'}`}</Snippet>}
+          />
+        </p>
+      </div>
+    );
+  }
+}
 
-export default connect(mapStateToProps)(PilApply);
+export default (PilApply);
```

Comment:
random `()`

Reply:
I don't actually understand why these (pil-apply, role-apply) are components - they are just single links shown depending on a couple of conditions, these should probably just be used in the main page component as is

Reply:
I don't understand it either - they were there, and knowing how rigorous the code reviews can be, I assumed that's right. I removed all the wrappers now.

Reply:
![](https://images-na.ssl-images-amazon.com/images/I/81mMtW%2BZbPL._SX425_.jpg)